http-server/src/main/java/io/airlift/http/server/HttpServerModule.java

```diff
@@ -57,32 +61,56 @@
-        binder.bind(HttpServerInfo.class).in(Scopes.SINGLETON);
-        // override with HttpServerBinder.enableVirtualThreads()
-        newOptionalBinder(binder, Key.get(Boolean.class, EnableVirtualThreads.class)).setDefault().toInstance(false);
-        // override with HttpServerBinder.enableLegacyUriCompliance()
-        newOptionalBinder(binder, Key.get(Boolean.class, EnableLegacyUriCompliance.class)).setDefault().toInstance(false);
-        newSetBinder(binder, Filter.class).addBinding()
-                .to(TracingServletFilter.class).in(Scopes.SINGLETON);
-        newSetBinder(binder, HttpResourceBinding.class);
-        newOptionalBinder(binder, SslContextFactory.Server.class);
-        newOptionalBinder(binder, Key.get(Boolean.class, EnableCaseSensitiveHeaderCache.class)).setDefault().toInstance(false);
+        binder.bind(qualifiedKey(HttpServer.class))
+                .toProvider(new HttpServerProvider(qualifier))
+                .in(SINGLETON);
+        newOptionalBinder(binder, qualifiedKey(ClientCertificate.class)).setDefault().toInstance(ClientCertificate.NONE);
+        newExporter(binder).export(qualifiedKey(HttpServer.class)).withGeneratedName();
+        newSetBinder(binder, qualifiedKey(ServerFeature.class));
+        newSetBinder(binder, qualifiedKey(Filter.class)).addBinding()
+                .to(TracingServletFilter.class)
+                .in(SINGLETON);
+        newSetBinder(binder, qualifiedKey(HttpResourceBinding.class));
+        newOptionalBinder(binder, qualifiedKey(SslContextFactory.Server.class));
+        configBinder(binder).bindConfig(qualifiedKey(HttpServerConfig.class), HttpServerConfig.class, configPrefix);
+        newOptionalBinder(binder, qualifiedKey(HttpsConfig.class));
+        newOptionalBinder(binder, qualifiedKey(ByteBufferPool.class));
 
-        configBinder(binder).bindConfig(HttpServerConfig.class);
-        newOptionalBinder(binder, HttpsConfig.class);
-        newOptionalBinder(binder, ByteBufferPool.class);
+        binder.bind(qualifiedKey(AnnouncementHttpServerInfo.class))
+                .toProvider(new AnnouncementHttpServerInfoProvider(qualifier))
+                .in(SINGLETON);
 
-        binder.bind(AnnouncementHttpServerInfo.class).to(LocalAnnouncementHttpServerInfo.class).in(Scopes.SINGLETON);
+        binder.bind(qualifiedKey(HttpServerInfo.class))
+                    .toProvider(new HttpServerInfoProvider(qualifier))
+                    .in(SINGLETON);
 
-        install(conditionalModule(HttpServerConfig.class, HttpServerConfig::isHttpsEnabled, moduleBinder ->
-                configBinder(moduleBinder).bindConfig(HttpsConfig.class)));
+        install(conditionalModule(qualifiedKey(HttpServerConfig.class), HttpServerConfig.class, configPrefix, HttpServerConfig::isHttpsEnabled, moduleBinder ->
+                configBinder(moduleBinder).bindConfig(qualifiedKey(HttpsConfig.class), HttpsConfig.class, configPrefix)));
```

Comment:
If we remove binding from conditional module, this becomes
```java
install(conditionalModule(HttpServerConfig.class, configPrefix, HttpServerConfig::isHttpsEnabled, moduleBinder ->
        configBinder(moduleBinder).bindConfig(qualifiedKey(HttpsConfig.class), HttpsConfig.class, configPrefix)));

configBinder(binder).bindConfig(qualifiedKey(HttpServerConfig.class), HttpServerConfig.class, configPrefix);
```
So one extra line here, but the code becomes more clear and explicit